File: DailyChallenges/CPP/LeetCode_1768.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n1 = word1.size(), n2 = word2.size();
        int n = min(n1, n2);
        string ans;
        for (int i=0; i < n; ++i){
            ans.push_back(word1[i]);
            ans.push_back(word2[i]);
        }
        return ans + ((n1<n2)? word2.substr(n, n2) : word1.substr(n,n1));
    }
};

File: DailyChallenges/Scala/LeetCode_1768.scala
object Solution {
    def mergeAlternately(word1: String, word2: String): String = {
        val n1 = word1.length
        val n2 = word2.length
        
        val reminder = if (n1 > n2) word1.slice(n2, n1) else word2.slice(n1, n2)
        word1.zip(word2).foldLeft(""){ (s, p) => s + p._1 + p._2 } + reminder
    }
}
